src/modules/sync/validation/entitySchema.ts

```diff
@@ -0,0 +1,164 @@
+  ContentTypeSyncModel,
+  z.ZodTypeDef,
+  { content_groups: ReadonlyArray<Replace<ContentTypeContracts.IContentTypeGroup, { codename: string }>> }
+> = z
+  .strictObject({ content_groups: z.array(ContentGroupSchema) })
+  .passthrough()
+  .transform(obj => ({ ...obj, groups_number: obj.content_groups.length === 0 ? "zero" : "multiple" }))
+  .pipe(
+    z.discriminatedUnion("groups_number", [
+      z.strictObject({
+        name: z.string(),
+        codename: z.string(),
+        groups_number: z.literal("zero"),
+        content_groups: z.array(ContentGroupSchema).length(0),
+        elements: z.array(TypeElementsSchemasUnion),
+      }),
+      z.strictObject({
+        name: z.string(),
+        codename: z.string(),
+        groups_number: z.literal("multiple"),
+        content_groups: z.array(ContentGroupSchema).min(1),
+        elements: z.array(TypeElementWithGroupSchemasUnion),
+      }),
+    ])
+      .refine(a =>
+        a.groups_number === "zero"
+          ? true
+          : a.elements.every(e => a.content_groups.map(c => c.codename).includes(e.content_group.codename)), {
+        message: "Content group codename must be one of the type's content group codename",
+      }),
+  )
+  .transform(obj => omit(obj, ["groups_number"]));
+
+export const WebSpotlightSchema: z.ZodType<WebSpotlightSyncModel> = z.strictObject({
+  enabled: z.boolean(),
+  root_type: CodenameReferenceSchema.nullable(),
+});
+
+export const CollectionSchema: z.ZodType<CollectionSyncModel> = z.strictObject({
+  name: z.string(),
+  codename: z.string(),
+});
+
+export const SpaceSchema: z.ZodType<SpaceSyncModel> = z.strictObject({
+  name: z.string(),
+  codename: z.string(),
+  web_spotlight_root_item: CodenameReferenceSchema.optional(),
+  collections: z.array(CodenameReferenceSchema),
+});
+
+export const LanguageSchema: z.ZodType<LanguageSyncModel> = z.discriminatedUnion("is_default", [
+  z.strictObject(
+    {
+      name: z.string(),
+      codename: z.string(),
+      is_active: z.boolean(),
+      is_default: z.literal(true),
+    },
+  ),
+  z.strictObject(
+    {
+      name: z.string(),
+      codename: z.string(),
+      is_active: z.boolean(),
+      is_default: z.literal(false),
+      fallback_language: CodenameReferenceSchema,
+    },
+  ),
+]);
+
+const WorkflowColorSchema = z.union([
+  z.literal("gray"),
+  z.literal("red"),
+  z.literal("rose"),
+  z.literal("light-purple"),
+  z.literal("dark-purple"),
+  z.literal("dark-blue"),
+  z.literal("light-blue"),
+  z.literal("sky-blue"),
+  z.literal("mint-green"),
+  z.literal("persian-green"),
+  z.literal("dark-green"),
+  z.literal("light-green"),
+  z.literal("yellow"),
+  z.literal("pink"),
+  z.literal("orange"),
+  z.literal("brown"),
+]);
+
+const WorkflowPublishedStepSchema = z.strictObject({
+  name: z.string(),
+  codename: z.string(),
+  create_new_version_role_ids: z.array(z.string()).length(0),
+  unpublish_role_ids: z.array(z.string()).length(0),
+});
```

Comment:
Use `satisfies` in all the `strictObject` schemas with all the properties set to required as that's the best way to ensure we don't forget even an optional property.